### PR TITLE
[WIP] support for HapiJs

### PR DESCRIPTION
**Pull request by emanuelet**
Adding export to connect the middleware in HapiJs.

As discussed in https://github.com/i18next/i18next-http-middleware/issues/48

#### Checklist

- [x] only relevant code is changed (make a diff before you submit the PR)
- [x] run tests `npm run test`
- [x] tests are included

#### Checklist (for documentation change)

- [x] only relevant documentation part is changed (make a diff before you submit the PR)
- [x] motivation/reason is provided

### CODE REVIEW

**Comment by emanuelet** [comment]
trying to get my head around why the plugin is not able to attach headers and what not to the response. If anyone has suggestions, please let me know

File: README.md [hunk 1]
```diff
@@ -88,6 +87,48 @@ app.get('myRoute', (request, reply) => {
...
+i18next.use(middleware.LanguageDetector).init({
+  preload: ['en', 'de', 'it'],
+  ...otherOptions
+})
+
+const server = Hapi.server({
+  port: port,
+  host: '0.0.0.0',
+
+const app = fastify()
```
Comment (adrai) [inline README.md:106]: this is probably not correct, right?

Comment (emanuelet) [reply README.md:106]: ah yes, will remove it now

File: lib/index.js [hunk 2]
```diff
@@ -101,9 +113,34 @@ export function handle (i18next, options = {}) {
...
+  register: (server, options) => {
+    options.attachLocals = true
+    const middleware = handle(options.i18next, {
+      ...options
+    })
+
+    server.ext('onPreHandler', (request, h) => {
+      middleware(
+        request,
+        h.response() || request.Response || request.response,
```
Comment (adrai) [inline lib/index.js:134]: This line probably needs to be just `request.raw.res` this will fix the header issue

Comment (emanuelet) [reply lib/index.js]: that worked, thanks! 👍

**Comment by adrai** [comment]
Thank you for your contribution, it has just been released in v3.2.0

**Pull request merged**